lua/entities/gmod_wire_expression2/core/strfunc.lua

```diff
@@ -12,24 +12,21 @@ local function checkFuncName( self, funcname )
 	end
 end
 
-registerCallback("construct", function(self) self.strfunc_cache = {} end)
+registerCallback("construct", function(self) self.strfunc_cache = {{}, {}} end)
 
 local insert = table.insert
 local concat = table.concat
 local function findFunc( self, funcname, typeids, typeids_str )
-	local func, func_return_type, vararg
-
-	self.prf = self.prf + 40
+	local func, func_return_type
 
 	local str = funcname .. "(" .. typeids_str .. ")"
-	for i=1,#self.strfunc_cache do
-		local t = self.strfunc_cache[i]
-		if t[1] == str then
-			return t[2], t[3], t[4]
-		end
+
+  local cached = self.strfunc_cache[1][str]
```

Comment:
Could self.strfunc_cache be set to a local variable? Or maybe even something like
```lua
local hash, limiter = self.strfunc_cache[1], self.strfunc_cache[2]
```
To avoid a billion table indexes